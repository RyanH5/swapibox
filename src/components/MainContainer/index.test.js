import React from 'react';
import { shallow } from 'enzyme';
import { MainContainer } from './index';

describe('MainContainer', () => {
  let wrapper;
  let mockCategoryData;
  let mockUpdateCards;
  let mockDisplayFavorites;

  beforeEach(() => {
    mockCategoryData = [{
      name: 'SandDigger',
      vehicleClass: 'wheeled',
      model: 't-990',
      passengers: 500
    }];

    mockUpdateCards = jest.fn();
    mockDisplayFavorites = jest.fn();
    wrapper = shallow(<MainContainer 
      mockCategoryData={mockCategoryData} 
      updateCards={mockUpdateCards}
      displayFavorites={mockDisplayFavorites}
    />);

  });
  it('should match snapshot with all data', () => {
    

    expect(wrapper).toMatchSnapshot();
  });

  it('should call updateCards on click', () => {
    wrapper.find('button').first().simulate('click');
    const expected = 'https://swapi.co/api/people/';

    expect(mockUpdateCards).toHaveBeenCalledWith(expected);
  });
});