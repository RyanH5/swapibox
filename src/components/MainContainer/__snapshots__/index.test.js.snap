// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContainer should match snapshot with all data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    displayFavorites={[Function]}
    mockCategoryData={
        Array [
            Object {
              "model": "t-990",
              "name": "SandDigger",
              "passengers": 500,
              "vehicleClass": "wheeled",
            },
          ]
    }
    updateCards={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        Array [],
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          Array [],
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
